model/cluster/block.go

```diff
@@ -70,11 +61,25 @@ func NewBlock(untrusted UntrustedBlock) (*Block, error) {
 //
 // This constructor must be used **only** for constructing the root block,
 // which is the only case where zero values are allowed.
-func NewRootBlock(untrusted UntrustedBlock) *Block {
-	return &Block{
-		Header:  untrusted.Header,
-		Payload: untrusted.Payload,
+func NewRootBlock(untrusted UntrustedBlock) (*Block, error) {
+	rootHeaderBody, err := flow.NewRootHeaderBody(flow.UntrustedHeaderBody(untrusted.Header))
+	if err != nil {
+		return nil, fmt.Errorf("invalid root header body: %w", err)
+	}
+
+	if rootHeaderBody.ParentID != flow.ZeroID {
+		return nil, fmt.Errorf("ParentID must be zero")
+	}
```

Comment:
```suggestion
```
This is not generally true -- only true for genesis blocks. Root blocks in a spork refer to a parent block from the previous spork.